lib/std/fs.zig

```diff
@@ -48,6 +48,27 @@ pub const MAX_PATH_BYTES = switch (builtin.os.tag) {
         @compileError("PATH_MAX not implemented for " ++ @tagName(builtin.os.tag)),
 };
 
+/// This represents the maximum size of a UTF-8 encoded file name component that the
```

Comment:
Windows already states that it depends on what the filesystem configures, so the more conservative approach is to say "what common filesystems support". 
The gnu libc docs also hint such behavior: `Portability Note: On some systems, the GNU C Library defines NAME_MAX, but does not actually enforce this limit. `
https://www.gnu.org/software/libc/manual/html_node/Limits-for-Files.html

I have no idea, if its a good idea to separate by name, because those are rare edge cases.
Hard guarantees require to ask the Kernel (compiletime/runtime configuration).